### PR TITLE
Api provides only Editor permission to publish articles

### PR DESCRIPTION
**Pull request by arunbhalli**
**PT-Story:**  https://www.pivotaltracker.com/story/show/178264656
### User Story 
``` 
As an Api
To maintain a high standard of service
we need to provide a method for only editors to publish articles
``` 
## Changes proposed in this pull request: 
* New role EDITOR  to USER in ENUM
* Add `published` boolean to Article migration to ARTICLE
* Sad path that journalist CANNOT publish
## What I have learned working on this feature: 
*   How to give permission to different roles

### CODE REVIEW

File: app/controllers/api/articles_controller.rb [hunk 1]
```diff
@@ -32,11 +32,21 @@ def create
 
   def update
     article = Article.find(params[:id])
-    updated_article = article.update(article_params)
-    if updated_article
-      render json: { message: 'Your article has been successfully updated!' }, status: 200
+    if params[:published]
+      if current_user&.editor?
```
Comment (1CIM) [inline app/controllers/api/articles_controller.rb:36]: nice to know :3

**Review by 1CIM — APPROVED** [review]
Looking nice! we just need to remember to format all the time so we don't get those bs file changes ✌️

**Review by DavDan1 — APPROVED** [review]

File: app/controllers/api/articles_controller.rb (hunk 1 above)
Comment (offcast-dev) [inline app/controllers/api/articles_controller.rb:36]: There is a lot of nested if statements here. I thought you can take `if current_user&.editor?` outside `if params[:published]`, so to check it first If true - you just update the article in the else you can checkif params have `:published`. Also It feels like there is a way to do it with case statements, but I don't know

**Review by offcast-dev — APPROVED** [review]
Can do a quick refactor refactor but not a blocker

File: app/controllers/api/articles_controller.rb (hunk 1 above)
Comment (LucasKnudsen) [reply app/controllers/api/articles_controller.rb:36]: Thought you would like that :(

**Pull request merged**